G1ANT.Addon.Selenium/Api/SeleniumWrapper.cs

```diff
@@ -393,27 +386,69 @@ public string GetOuterHtml(SeleniumCommandArguments search)
 
         public string GetTextValue(SeleniumCommandArguments search, TimeSpan timeout)
         {
-            PreCheckCurrentWindowHandle();
-            if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
-                webDriver.SwitchTo().Frame(FindElement(search.IFrameSearch.Value, search.IFrameBy.Value, timeout));
-            var element = FindElement(search.Search.Value, search.By.Value, timeout);
+            var element = GetElementInFrame(search, timeout);
+
             var res = element?.Text ?? string.Empty;
             if (!string.IsNullOrEmpty(search.IFrameSearch?.Value))
                 webDriver.SwitchTo().DefaultContent();
             return res;
         }
 
+        public DataTable GetTableElement(SeleniumCommandArguments search, TimeSpan timeout)
+        {
+            var element = GetElementInFrame(search, timeout);
+
+            if (element == null)
+                throw new Exception("Cannot find the HTML element. Try to change the search phrase or \"by\" argument value so that the correct element is found");
+            else if (element.TagName != "table")
+                throw new Exception($"The element found has the \"{element.TagName}\" tag name. Try to change the search phrase so that the \"table\" element is found instead");
+
+            var dataTable = new DataTable();
+            var trElements = element.FindElements(By.TagName("tr"));
+            dataTable = AddColumnNamesFromThElements(dataTable, trElements[0].FindElements(By.TagName("th")));
+
+            foreach (var trElement in trElements)
+            {
+                var tdElements = trElement.FindElements(By.TagName("td"));
+                dataTable = AddColumnsIfThereAreMoreTdElements(dataTable, tdElements.Count);
+
+                dataTable.Rows.Add(tdElements.Select(td => td.Text).ToArray());
```

Comment:
take into account colspan property of the td tag to add cell into the correct column

Reply:
and what about the rowspan property, it should also be handled